File: tests/test_waveform_from_encoded_bytes.rs
mod fixtures;

mod test_waveform_from_encoded_bytes {
    use crate::fixtures::*;
    use babycat::Waveform;

    #[test]
    fn test_circus_of_freaks_default_1() {
        let bytes = std::fs::read(COF_FILENAME).unwrap();
        let waveform = Waveform::from_encoded_bytes(&bytes, Default::default()).unwrap();
        assert_eq!(waveform.num_channels(), COF_NUM_CHANNELS);
        assert_eq!(waveform.num_frames(), COF_NUM_FRAMES);
        assert_eq!(waveform.frame_rate_hz(), COF_FRAME_RATE_HZ)
    }

    #[test]
    fn test_invalid_bytes_1() {
        let bytes = "asdfasdfasdfe".to_string().into_bytes();
        let err = Waveform::from_encoded_bytes(&bytes, Default::default()).unwrap_err();
        assert_eq!(err.error_type(), "UnknownInputEncoding")
    }
}

File: tests-python/test_waveform_from_encoded_bytes.py
"""
Tests loading waveform from encoded bytes.

These tests mirror the ones in ``../tests/test_waveform_from_encoded_bytes.rs``
"""
import pytest
from fixtures import *

import babycat

Waveform = babycat.Waveform
bexc = babycat.exceptions


with open(COF_FILENAME, "rb") as fh:
    COF_BYTES = fh.read()


def test_circus_of_freaks_default_1():
    waveform = Waveform.from_encoded_bytes(COF_BYTES)
    assert waveform.num_channels == COF_NUM_CHANNELS
    assert waveform.num_frames == COF_NUM_FRAMES
    assert waveform.frame_rate_hz == COF_FRAME_RATE_HZ


def test_invalid_bytes_1():
    with pytest.raises(bexc.UnknownInputEncoding):
        Waveform.from_encoded_bytes(b"asdfasdfasdfe")
